ONSdigital/census-rm-exception-manager-2021
# Allow retry before logging & filter exceptions based on seen count

# Motivation and Context
Sometimes, the 3 retries with a 1-second delay is inadequate, and we end up logging and raising an exception with the exception manager. This could be really annoying during the Census, because it will generate a lot of noise. When we've been war-gaming, we've seen that a flood of messages can cause exceptions which resolve themselves, due to locking contention, for example.

# What has changed
Implemented a configurable "minimum number of times to see an exception before we log it".

Implemented a filter on the admin API so that it only returns exceptions which have been seen a certain number of times.

ADDENDUM: I kinda went a bit mad and added more features. Can now "throw away" bad messages (silently skip, don't quarantine, don't log). Can suppress all logging (but not skip/quarantine) if logs are getting noisy. Also, added an expiry date for rules.

# How to test?
Create some exceptions. Only the ones which have been seen more than once should log, and the toolbox bad message wizard should also not see anything which has only been seen once.

# Links
Trello: https://trello.com/c/hP1iPanw
